rubicon_ml/viz/experiments_table.py

```diff
@@ -0,0 +1,281 @@
+import dash_bootstrap_components as dbc
+from dash import dash_table, dcc, html
+from dash.dependencies import ALL, Input, Output, State
+
+from rubicon_ml.viz.base import VizBase
+from rubicon_ml.viz.colors import light_blue, plot_background_blue
+
+
+class ExperimentsTable(VizBase):
+    def __init__(self, experiments, is_selectable=True, dash_kwargs={}):
+        super().__init__(dash_kwargs=dash_kwargs, dash_title="rubicon-ml: experiment table")
+
+        self.experiments = experiments
+        self.is_selectable = is_selectable
+
+        self.commit_hash = None
+        self.experiment_records = []
+        self.github_url = None
+        self.metric_names = set()
+        self.parameter_names = set()
+        self.all_columns = ["id", "name", "created_at", "model_name", "commit_hash", "tags"]
+        self.hidden_columns = []
+
+        commit_hashes = set()
+        show_columns = {"id", "created_at"}
+
+        for experiment in self.experiments:
+            if experiment.commit_hash is not None:
+                commit_hashes.add(experiment.commit_hash)
+                show_columns.add("commit_hash")
+
+            if experiment.model_name is not None:
```

Comment:
In the notebook, the "Model_name" column was included in the table when selecting "show all" despite the values all being empty. Should the empty columns be omitted from "show all" as well?

Reply:
yeah I was debating that one as well. when the widget loads, `model_name` is hidden (as well as any other totally null column). I originally had show all only show the ones that weren't null, but then I thought "that isn't really _all_, is it?" so I decided that the show all button would literally be all regardless of the values. I'm definitely not tied to that idea if you think it'd make more sense to only show the non-null ones

Reply:
Ok that makes sense! I think it should be fine to just leave. We could also remove the null columns from the table and just have the `toggle columns` drop down have those selected with (N/A) or (null) next to them? For example, when the option `model_name` has no values, the toggle drop down could have it checked when selecting show all but change it to `model_name (N/A)`. Or we could explicitly say "null" in each of the cells for model_name so that it's more readable and the users can understand that it's not an error. What do you think would be best?

Reply:
for now, I'm just gonna leave it as is. I just double checked what the current dashboard does, and it shows empty ones. so we'll leave them the same for now. those are all good ideas though, I'd like to get some user feedback to see which they'd prefer